crates/benchmark-runner/src/guest_programs.rs

```diff
@@ -0,0 +1,105 @@
+//! Guest program input generation and metadata types
+
+use guest_libs::BincodeBlock;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::path::Path;
+use walkdir::WalkDir;
+use witness_generator::BlockAndWitness;
+use zkvm_interface::Input;
+
+/// Metadata trait for guest program inputs
+pub trait GuestInputMetadata: Serialize + DeserializeOwned + Clone + Send + Sync {}
+impl GuestInputMetadata for () {}
+
+/// Represents a guest program input with associated metadata
+#[derive(Debug, Clone)]
```

Comment:
This was 80% of the need of the ere PR. The other 20% is Send+Sync for rayon at the benchmark-runner level.